lib/app.js

```diff
@@ -0,0 +1,78 @@
+const express = require('express');
+const cors = require('cors');
+const client = require('./client.js');
+const app = express();
+const morgan = require('morgan');
+const ensureAuth = require('./auth/ensure-auth');
+const createAuthRoutes = require('./auth/create-auth-routes');
+
+app.use(cors());
+app.use(express.json());
+app.use(express.urlencoded({ extended: false }));
+app.use(morgan('dev')); // http logging
+
+const authRoutes = createAuthRoutes();
+
+// setup authentication routes to give user an auth token
+// creates a /auth/signin and a /auth/signup POST route. 
+// each requires a POST body with a .email and a .password
+app.use('/auth', authRoutes);
+
+// everything that starts with "/api" below here requires an auth token!
+app.use('/api', ensureAuth);
+
+// and now every request that has a token in the Authorization header will have a `req.userId` property for us to see who's talking
+app.get('/api/test', (req, res) => {
+  res.json({
+    message: `in this proctected route, we get the user's id like so: ${req.userId}`
+  });
+});
+///////////////////////////////////get plants by id
+app.get('/api/todo', async(req, res) => {
+  try {
+    const data = await client.query('SELECT * FROM todo WHERE todo.owner_id = $1', [req.userId]);
+    
+    res.json(data.rows[0]);
+  } catch(e) {
+    
+    res.status(500).json({ error: e.message });
+  }
+});
+////////////////////////////////put plants by id
+app.put('/api/todo/:id', async(req, res) => {
+  try {
+    const data = await client.query(`
+    UPDATE todo
+    SET done = true
+    WHERE todo.owner_id = $1
+    and todo.id = $2
+    RETURNING *;
+    `[req.userId, req.params.id ]);
```

Comment:
you'll need a comma between the back-tic and the opening bracket.  Should look like: 
``` 
`, [
```

Reply:
thank you